packages/clay-datepicker/src/ClayDatepicker.js

```diff
@@ -0,0 +1,384 @@
+	 */
+	const elementRef = useRef(null);
+
+	/**
+	 * Handles the click on element of the day
+	 * @param {!Date} date
+	 */
+	function handleDayClicked(date) {
+		setDaySelected(date);
+		setInputValue(date);
+	}
+
+	/**
+	 * Handles the input change.
+	 * @param {!Event} event
+	 */
+	function inputChange(event) {
+		const {value} = event.target;
+		const format = `${dateFormat} ${time ? timeFormat : ''}`;
+		const date = moment(value, format);
+
+		if (date.isValid()) {
+			setCurrentMonth(date.toDate());
+			setDaySelected(date.toDate());
+
+			if (time) {
+				setCurrentTime(date.hours(), date.minutes());
+			}
+		}
+
+		if (onValueChange) {
+			onValueChange(value);
+		}
+
+		setInputValue(value);
+	}
+
+	/**
+	 * Handles datepicker view
+	 */
+	function handleCalendarButtonClicked() {
+		setExpanded(!expanded);
+	}
+
+	useEffect(
+		() => {
+			// It is not necessary to update the weeks when the native
+			// date is enabled. Unnecessary rendering and processing.
+			if (!useNative) {
+				setWeeks(currentMonth, firstDayOfWeek);
+			}
+		},
+		[currentMonth]
+	);
+
+	const classNames = classnames('input-group', {
+		[id]: id,
+	});
+
+	return (
+		<div className={classNames} ref={elementRef}>
+			<div className="input-group-item">
+				<InputDate
+					ariaLabel={ariaLabel}
+					currentTime={currentTime}
+					dateFormat={dateFormat}
+					defaultValue={defaultValue}
+					inputName={inputName}
+					onChange={inputChange}
+					placeholder={placeholder}
+					time={time}
+					timeFormat={timeFormat}
+					useNative={useNative}
+					value={inputValue}
+				/>
+				{!useNative && (
+					<div className="input-group-inset-item input-group-inset-item-after">
+						<Button
+							className="d-none d-md-inline-block"
+							icon="calendar"
+							onClick={handleCalendarButtonClicked}
+							spritemap={spritemap}
+							style="unstyled"
+						/>
+					</div>
+				)}
+			</div>
+			{!useNative && (
+				<Dropdown
+					containerRef={elementRef}
+					expanded={expanded}
+					onDocumentClick={() => setExpanded(false)}
+				>
+					<React.Fragment>
+						<div className="datepicker">
+							<DateNavigation
+								currentMonth={currentMonth}
+								months={months}
+								onDotClicked={() => {
+									setCurrentMonth(initialMonth);
+									setDaySelected(initialMonth);
+									setInputValue(initialMonth);
+								}}
+								onMonthChange={setCurrentMonth}
+								spritemap={spritemap}
+								years={years}
+							/>
+							<Weekdays
+								firstDayOfWeek={firstDayOfWeek}
+								weekdaysShort={weekdaysShort}
+							>
+								{({weekday, key}) => (
+									<Week key={key} weekday={weekday} />
+								)}
+							</Weekdays>
+							<div className="datepicker-body">
+								<Weekday weeks={weeks}>
+									{({day, key}) => (
+										<Day
+											day={day}
+											daySelected={daySelected}
+											key={key}
+											onClick={handleDayClicked}
+										/>
+									)}
+								</Weekday>
```

Comment:
The naming of `<Weekdays />` and `<Weekday />` here is a little confusing because they both render plural things. Is one of these a header of and the other is the list of dates?

Reply:
Yes. This can be a bit confusing, do you have any suggestions?

- `<Weekdays />` - Lists weekdays short
- `<Weekday />` - Lists the days of the month

Reply:
Something like:

- `<WeekdayHeader />` : for the list of "S M T W T F S" (currently, this is `<Weekdays />`).
- `<Weekday />`: for each of "S", "M" etc (currently this is `<Week />`; super confusing!).
- `<DaysRow />`:  for each row of "1 2 3 4 etc" (currently this is `<Weekday />`; again pretty confusing for me).
- `<DayNumber />`, maybe: for each of "1", "2" etc (currently this is `<Day />`).

Maybe not those exact names, but in a case like this where all the terms are very similar and repetitive, I think it's better to bias towards names that are longer and more literal and descriptive, otherwise it is easy to get confused.

Reply:
Thanks a lot for your suggestions, I like these names but only I would change `<DaysRow />` to `<DaysTable />` since it renders the table in rows and columns, what do you think?